DOC-7628 analytics examples for scope/collection

osfameron opened the pull request:
Please advise in particular if  the usage for Scope looks correct.
This is what I've inferred from SDK RFCs and seems to work for Java and dotNet SDKs, but would love a sanity check that I've got the best practice/intent correct.

osfameron committed:
DOC-7628 clean up Analytics.cs

Make the example run, as follows.

	cd docs-sdk-dotnet/modules/devguide/examples/dotnet
	# edit Program.cs to uncomment the Analytics.cs example
	dotnet run

osfameron committed:
DOC-7628 analytics examples for collection and scope

jeffrymorris commented on modules/devguide/examples/dotnet/Analytics.cs:
```diff
@@ -14,13 +14,45 @@ private class TestRequest
+                Console.WriteLine("\nHandle Scope");
+                // tag::handle-scope[]
+                IBucket bucket = await cluster.BucketAsync("travel-sample").ConfigureAwait(false);
+                var scope = bucket.Scope("inventory");
```
"Scope" is now obsolete, you should do a "await ScopeAsync" instead.

osfameron committed:
DOC-7628 correct ScopeAsync, and update analytics dataset

 * use ScopeAsync instead of deprecated .Scope method
 ** bump version

 * update creation of dataset as per
   https://blog.couchbase.com/collections-simplify-your-analysis-with-couchbase-analytics/

 ** this now creates the dataset as "`bucket`.`scope`" instead of "`bucket.scope`"
 ** as the convenience scope.AnalyticsQuery call seemed to work previously in the
    dotNet SDK, this suggests the previous behaviour was incorrect?
    Now, the handle-scope example (which previously ran) gives the error:
      `Unhandled exception. Couchbase.CouchbaseException: Analytics query failed: 24045`

 ** Commented out that section of code and accompanying documentation for now.
 ** will also raise a bug/clarification ticket with sdk team

osfameron commented:
Sorry for noise on this branch! I've reset to 501e5867c581975ad378f52403e23ca0ff1fbc6b which should be the last commit in this PR, @jeffrymorris can you confirm if that addresses your ScopeAsync comment and OK to merge?

RichardSmedley reviewed (approved):
Let's get this merged - lack of objection can be taken as approval,
in any case we need to move forward.
